app/src/test/java/io/gnosis/safe/ui/safe/add/AddSafeNameViewModelTest.kt

```diff
@@ -0,0 +1,119 @@
+package io.gnosis.safe.ui.safe.add
+
+import io.gnosis.data.models.Safe
+import io.gnosis.data.repositories.SafeRepository
+import io.gnosis.safe.TestLifecycleRule
+import io.gnosis.safe.appDispatchers
+import io.gnosis.safe.di.Repositories
+import io.gnosis.safe.test
+import io.gnosis.safe.ui.base.BaseStateViewModel
+import io.mockk.*
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.test.TestCoroutineDispatcher
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.setMain
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import pm.gnosis.utils.asEthereumAddress
+import java.lang.IllegalStateException
+
+class AddSafeNameViewModelTest {
+
+    @get:Rule
+    val instantExecutorRule = TestLifecycleRule()
+
+    private val safeRepository = mockk<SafeRepository>()
+    private val repositories = mockk<Repositories>().apply {
+        every { safeRepository() } returns safeRepository
+    }
+
+    private lateinit var viewModel: AddSafeNameViewModel
+
+    @Before
+    fun setup() {
+        viewModel = AddSafeNameViewModel(repositories, appDispatchers)
+        Dispatchers.setMain(TestCoroutineDispatcher())
```

Comment:
Would it make sense to have a base class for the dispatchers setup?

Reply:
Yeah, I think so 👍